Tools/Scripts/webkitdirs.pm

```diff
@@ -3213,6 +3283,17 @@ sub mobileSafariBundle()
     return installedMobileSafariBundle();
 }
 
+sub visionSafariBundle()
+{
+    determineConfigurationProductDir();
+
+    # Use MobileSafari.app in product directory if present.
+    if (isVisionOSWebKit() && -d "$configurationProductDir/MobileSafari.app") {
+        return "$configurationProductDir/MobileSafari.app";
+    }
+    return installedVisionSafariBundle();
+}
```

Comment:
Here's an example where the code duplication could be avoided with some refactoring. If you changed `mobileSafariBundle()` to test for `if (isIOSWebKit() || isVisionOSWebKit() ...)`, then you could share the function.